File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        String temp = Integer.toString(n, k);
        
        String[] arr = temp.split("0");
        
        for (String data : arr){

            if (data.equals("")) continue;
            long num = Long.parseLong(data);
            if (isPrime(num)) answer++;
        }
        
        
        return answer;
    }
    public boolean isPrime(long num) {
        if (num < 2) return false;
        
        for (int i = 2; i<=Math.sqrt(num);i++){
            if (num % i == 0) return false;
        }
        return true;
    }
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.cpp
#include <bits/stdc++.h>

using namespace std;

string s,temp;

bool isPrime(string s_num) {
    long long num = stoll(s_num);
    if (num == 1) return false;

    long long end = sqrt(num);

    for (int i=2;i<=end;i++) {
        if (num % i == 0) return false;
    }

    return true;
}

int solution(int n, int k) {
    int answer = 0;

    while (n > 0) {
        s += to_string(n % k);
        n /= k;
    }

    reverse(s.begin(), s.end());

    for (int i=0;i<s.size();i++) {
        if (s[i] == '0') {
            if (temp.size() > 0 && isPrime(temp)) answer++;
            temp = "";
        }
        else {
            temp += s[i];
        }
    }

    if (temp.size()) {
        if (isPrime(temp)) answer++;
    }

    return answer;
}